pkg/controller/kieapp/kieapp_controller.go

```diff
@@ -769,3 +777,89 @@ func (reconciler *Reconciler) createConfigMap(obj v1.OpenShiftObject) (*corev1.C
+	if err := reconciler.Service.List(context.TODO(), listOps, cmList); err != nil {
+		log.Warn("Failed to list ConfigMaps. ", err)
+	} else {
+		serverDcList := make(map[string]int32)
+		for _, server := range env.Servers {
+			for _, sDc := range server.DeploymentConfigs {
+				serverDcList[sDc.Name] = sDc.Spec.Replicas
+			}
+		}
+		for _, cm := range cmList.Items {
+			for _, ownerRef := range cm.OwnerReferences {
+				if ownerRef.Kind == "DeploymentConfig" && cm.Labels[constants.KieServerCMLabel] != "" && cm.Labels[constants.KieServerCMLabel] != "DETACHED" {
+					// if replicas greater than zero, execute PUT against console
+					if serverDcList[ownerRef.Name] != 0 {
+						reconciler.reqServerIDinConsole(instance, env.Console, cm.Labels["services.server.kie.org/kie-server-id"], "PUT")
+					} else {
+						// if replicas equal zero, execute DELETE against console
+						dcObj := &oappsv1.DeploymentConfig{}
+						if err := reconciler.Service.Get(context.TODO(), types.NamespacedName{Name: ownerRef.Name, Namespace: cm.Namespace}, dcObj); err != nil {
+							log.Error(err)
+						}
+						if serverDcList[ownerRef.Name] == dcObj.Status.AvailableReplicas {
+							cmObj := &corev1.ConfigMap{}
+							if err := reconciler.Service.Get(context.TODO(), types.NamespacedName{Name: ownerRef.Name, Namespace: cm.Namespace}, cmObj); err != nil {
+								log.Error(err)
+							}
+							cmObj.Labels[constants.KieServerCMLabel] = "DETACHED"
+							log.Infof("%s replicas set to zero so relabeling associated ConfigMap as DETACHED", cm.Name)
+							if _, err = reconciler.UpdateObj(cmObj); err != nil {
+								log.Error(err)
+							} else {
+								reconciler.reqServerIDinConsole(instance, env.Console, cm.Labels["services.server.kie.org/kie-server-id"], "DELETE")
+							}
+						}
+					}
+				}
+			}
+		}
+	}
+}
+
+// reqServerIDinConsole sends a http request to the console, if deployed, using the specified method type
+func (reconciler *Reconciler) reqServerIDinConsole(instance *v1.KieApp, console v1.CustomObject, serverID, method string) {
+	if len(console.Services) > 0 {
+		for _, service := range console.Services {
+			svcObj := &corev1.Service{}
+			if err := reconciler.Service.Get(context.TODO(), types.NamespacedName{Name: service.Name, Namespace: instance.Namespace}, svcObj); err != nil {
+				log.Error(err)
+			}
+			for _, port := range svcObj.Spec.Ports {
+				if port.Name == "http" {
+					// Create client
+					client := &http.Client{}
+
+					url := fmt.Sprintf("%s://%s:%d/%s/%s", port.Name, svcObj.Spec.ClusterIP, port.Port, "rest/controller/server", serverID)
+					log.Debug(method, " ", url)
+					// Create request
+					req, err := http.NewRequest(method, url, nil)
+					if err != nil {
+						log.Error(err)
+					}
+
+					// Fetch Request
+					req.Header.Set("Content-Type", "application/json")
+					req.SetBasicAuth(instance.Spec.CommonConfig.AdminUser, instance.Spec.CommonConfig.AdminPassword)
+					resp, err := client.Do(req)
```

Comment:
@rhtevan this goes against our dependency hierarchy. The operator is an infrastructure component that only interacts with Kubernetes resources and the k8s API. BC should watch or poll the kube API as needed.

Reply:
@rhtevan i strongly agree and the more i test, i'm finding certain scenarios are crashing the operator. i think i'd be ok with the labels logic if absolutely necessary, but i'd like to at least remove the BC api logic from this PR.

Reply:
@tchughesiv 
If it is the case, then we should take the BC logic out. Please keep the labeling logic. I'll explore alternatives to tackle.